pkg/apis/rook.io/v1alpha2/types.go

```diff
@@ -81,19 +81,11 @@ type Placement struct {
 
 type ResourceSpec map[string]v1.ResourceRequirements
 
-type NetworkSpec struct {
-	metav1.TypeMeta `json:",inline"`
-
-	// HostNetwork to enable host network
-	HostNetwork bool `json:"hostNetwork"`
+type NetworkSelector string
```

Comment:
Why define a NetworkSelector? It's more readable to me if we just use the string directly.

Reply:
Previously I thought there are ways to separately validate each different network providers selector into more specific type. But I think that's a bad idea now.